internal/otel/config/helpers/receivers/prometheus_receiver.go

```diff
@@ -46,6 +48,9 @@ func PrometheusReceiverCfg() *PrometheusConfig {
 				- localhost:9090
 	*/
 
+	if metricsPort == 0 {
```

Comment:
if we're doing this again here (same) as telemetry, we should probably be doing this once at the start somewhere to avoid this duplication (relates to comment mentioned in telemetry.go). Maybe at the provider is the wrong level but somewhere earlier :thinking: